### DISCUSSION TITLE
Why do you love Laravel?

### DISCUSSION THREAD

**Discussion by imliam** [laravel/framework#32759] [General]
The question is simple - **why do you love Laravel**?!

At one point, something attracted you to Laravel, and you're still here. What is it that you find so attractive about this framework?

I ask this because there's so much drama and judgement out there about PHP and Laravel. It's not uncommon to see people spouting "PHP is a dead language", "why don't you just use Java?", "PHP isn't as fast as Golang", "Laravel makes you use bad practices" and other things

I think there's plenty of reasons to stick with PHP and Laravel, despite knowing other programming languages, frameworks and ecosystems. I'd love to see other people's reactions to this question.

**Comment by Cannonb4ll** [comment]
Because of the simplicity and easy syntax 🚀

And most of all, because of the huge community which is growing each day.

**Comment by imliam** [comment]
The big one for me is the community. Somehow, Taylor has managed to cultivate and craft a community who shares the same values, and it's lovely to be around:

**Sharing** is a huge part, whether it be open-sourcing their own code, giving tips or tricks they've picked up on or anything else they might have to give back that can benefit other people.

**Teaching** is another - the documentation for tools in the Laravel ecosystem is some of the best, there are tens-of-thousands of tutorials about the framework publicly available, and people are always so keen to help new people get on board.

**Clean code** is yet another part of that. People don't just try to get stuff done in the hackiest way possible all the time, but there is genuinely nice-to-use and nice-to-read code all over the place, and a big part of the community benefits from that.

This has created an atmosphere where there are dedicated packages, tutorials, tools, etc. available for practically anything imaginable, they're all great to use because people care about teaching others how to use them, and I think it's great.

**Comment by ahinkle** [comment]
Oh, where do I begin?

I feel like I'm crafting some artwork 🎨 when working in Laravel. The syntax and documentation are so rich and elegant. Laravel achieves a high level of code maintainability and developer experience. It also features great tools such as Dusk, Valet, and Cashier to name a few.

**Comment by danharrin** [comment]
Personally, I enjoy working with the project structure that Laravel provides. It ensures my code is written consistently, is readable, and follows good industry practices. 🚀

**Comment by innoflash** [comment]
What else should i be using if not Laravel?

Personally i have got to understand deep OOP for any language with Laravel, this framework is just beautiful

**Comment by patrickbrouwers** [comment]
For me it's developer happiness. Writing code in Laravel just makes me happy. It's easy and elegant, but still powerful.

**Comment by jadonbrownza** [comment]
You can see the effort put into the framework to make it developer-friendly. I love the consistency. I love that Laravel chose PHP. Man, Laravel prevents issues that I didn't even know were a thing - hence the learning you get from just using the framework is amazing!

**Reply by jadonbrownza** [reply]
And also, because @JeffreyWay :)

**Comment by lpeterke** [comment]
I think I like the fact that I can get something up and running really quickly, no matter what it is. Authentication, Notification, Mails... It's all there and really easy to prototype things and find out. 

Sure, it can be a bit overwhelming to dive into Laravel today compared to a few years back (especially with all the additional tools and services like Nova, Spark, Vapor, etc.), but everything is so well documented and the community is so nice... working in the ecosystem is really encouraging in every aspect.

**Comment by shadoWalker89** [comment]
Laravel makes everything easier (great features and syntax).
Amazing documentation
Amazing community
And the most important of all it makes me a better developper, since i started using laravel 3 my thinking and my skills improved a lot, i enjoy deep diving into laravel source code and learn from it.

And last but not least @JeffreyWay i've been following and learning from this legend since his days in  nettuts and i learned about laravel from one of his videos on nettuts. @JeffreyWay is with no doubt the best php teacher i have ever seen and i'm so happy to have him into the laravel world so we can keep learning from him.

**Comment by DaviMenezes** [comment]
you want to see the circus catch fire right? humm

**Comment by horaceho** [comment]
The architecture. The "Architecture Concepts" section of the documentation is impressive. 

Eloquent, I have been used ActiveRecord and love Eloquent a lot.

The community and great collection of packages make my life a lot easier.

Thank you!

**Comment by pushpak1300** [comment]
Laracast and Eloquent!

**Reply by voidgraphics** [reply]
This! These two are the reason it stuck with me in the beginning.

**Reply by DaviMenezes** [reply]
@voidgraphics the beginning? No more?

**Reply by voidgraphics** [reply]
I was a young student, just starting to learn OOP and how to write SQL queries. So these two were like magic - everything suddenly became much easier to deal with.

**Comment by ivankolodii** [comment]
- Great documentation
- ecosystem(horizon,  forge, vapor, etc.)
- regular updates
- great community(Spatie, @JeffreyWay and many many other)
- a huge amount of packages

Our team uses Laravel as the main backend framework and totally  ❤️ it.
Thank you!

**Comment by elbojoloco** [comment]
It makes my life so much easier and my job so much more enjoyable.

**Comment by n123456789k** [comment]
<img src="https://i.chzbgr.com/original/8257536000/h342CBAB8/cheezburger-image-8257536000">

**Comment by mfn** [comment]
I once read about https://www.reddit.com/r/laravel/comments/ak8lz4/laravel_is_such_a_ravioli/ and realized, this is the very exact reason "Laravel works for me". Not for everyone though.

It usually finds a fine line between well architectured (composability, overrideability) and still being pragmatic/not feeling overengineered.
